src/test/TemplateProcessor.test.js

```diff
@@ -901,39 +906,53 @@ test("Solution Environment Files", async () => {
     expect(tp.output).toEqual(expected);
 });
 
-test("remove all DEFAULT_FUNCTIONS", async () => {
-    let template = {"fetchFunctionShouldNotExists": "${$fetch('https://raw.githubusercontent.com/geoffhendrey/jsonataplay/main/foobar.json')}"};
-    TemplateProcessor.DEFAULT_FUNCTIONS = {};
-    const tp = new TemplateProcessor(template);
-    await tp.initialize();
-    expect(tp.output).toStrictEqual({
-        "fetchFunctionShouldNotExists": {
-            "error": {
-                "message": "Attempted to invoke a non-function",
-                "name": "JSONata evaluation exception"
+describe("Testing DEFAULT_FUNCTIONS manipulation", () => {
+    let originalDefaultFunctions;
+
+    beforeEach(() => {
+        // Save the global variable before the test
+        originalDefaultFunctions = TemplateProcessor.DEFAULT_FUNCTIONS;
```

Comment:
originalDefaultFunctions = [...TemplateProcessor.DEFAULT_FUNCTIONS]
If you don't do this you are still allowing the content of DEFAULT_FUNCTIONS array to be destroyed